### PR TITLE
[rfxcom] Handler factory activation simplified

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java [hunk 1]
```diff
@@ -46,25 +47,22 @@
...
 
-    @Reference
-    protected void setSerialPortManager(final SerialPortManager serialPortManager) {
-        this.serialPortManager = serialPortManager;
-    }
+    private final SerialPortManager serialPortManager;
 
-    protected void unsetSerialPortManager(final SerialPortManager serialPortManager) {
-        this.serialPortManager = null;
+    @Activate
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java:63]: Afaik, "Activate" should only be added to an activate method, but not to the constructor.

Comment (lolodomo) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java:63]: This is done like that everywhere I believe.

Comment (lolodomo) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java:63]: For example: https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosHandlerFactory.java#L73

Comment (kaikreuzer) [reply bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java:63]: Ok, seems a legitimate case: https://docs.osgi.org/javadoc/osgi.cmpn/7.0.0/org/osgi/service/component/annotations/Activate.html

**Pull request merged**